packages/effect/test/RateLimiter.test.ts

```diff
@@ -1,223 +1,271 @@
-import { Clock, Deferred, Effect, Either, Fiber, pipe, Ref, TestClock } from "effect"
+import { Clock, Deferred, Effect, Either, Fiber, Option, RateLimiter, ReadonlyArray, Ref, TestClock } from "effect"
 import * as it from "effect-test/utils/extend"
-import { none } from "effect/Option"
-import * as RateLimiter from "effect/RateLimiter"
 import { assert, describe } from "vitest"
 
-describe("RateLimiterSpec", () => {
-  it.effect("execute up to max calls immediately", () => {
+describe("RateLimiter", () => {
+  describe("fixed-window", () => {
+    RateLimiterTestSuite("fixed-window")
+
+    it.effect("uses the token-bucket algorithm for token replenishment", () =>
+      Effect.scoped(Effect.gen(function*(_) {
+        // The limiter below should allow be to execute 10 requests immediately,
+        // prevent further requests from being executed, and then after 100 ms
+        // allow execution of another request.
+        const limit = yield* _(RateLimiter.make({
+          limit: 10,
+          interval: "1 seconds",
+          algorithm: "token-bucket"
+        }))
+        const deferred = yield* _(Deferred.make<void>())
+
+        // Use up all of the available tokens
+        yield* _(Effect.forEach(ReadonlyArray.range(1, 10), () => limit(Effect.unit)))
+
+        // Make an additional request when there are no tokens available
+        yield* _(
+          limit(Effect.unit),
+          Effect.zipRight(Deferred.succeed(deferred, void 0)),
+          Effect.fork
+        )
+        assert.isFalse(yield* _(Deferred.isDone(deferred)))
```

Comment:
You will technically need a yieldNow before this, as forks don't execute immediately.

Reply:
Good call - fixed 👍